R/check_gene_expression.R

```diff
@@ -0,0 +1,111 @@
+#'   \item{patient_id}{The anonymized unique identifier for this patient. For BC samples, this will be Res ID.}
+#'   \item{seq_type}{The assay type used to produce this data (will always be "mrna" in this case)}
+#'   \item{protocol}{Specifies the RNA-seq library construction protocol.}
+#'   \item{ffpe_or_frozen}{Specifies the way the source of nucleic acids was preserved. Either FFPE or frozen.}}
+#' 
+check_gene_expression = function(verbose=F,show_linkages=F){
+  # We start with the minimal metadata that corresponds to the tidy expression file (sample_metadata.tsv)
+  metadata_file = check_config_value(config::get("results_merged")$tidy_expression_metadata)
+  metadata_file = paste0(check_config_value(config::get("project_base")),metadata_file)
+  expression_data_rows = suppressMessages(read_tsv(metadata_file)) %>% 
+                                            dplyr::select(sample_id,tissue_status,seq_type,patient_id,biopsy_id,protocol,ffpe_or_frozen,cohort) %>%
+                                            dplyr::rename(c("mrna_sample_id"="sample_id"))
+  
+
+  
+  collapse_duplicates = TRUE
+
+  if(verbose){
+    nrows = nrow(expression_data_rows)
+    print(paste("starting with",nrows,"rows"))
+  }
+
+  
+  mark_duplicates_biopsy = function(ew){
+    mutate(expression_data_rows, sample_seqType = paste(biopsy_id, seq_type)) %>% 
+      filter(!is.na(mrna_sample_id)) %>% 
+      with( split(mrna_sample_id, sample_seqType) ) %>% 
+      .[lengths(.) > 1] %>% 
+      lapply(unique) %>% 
+      .[lengths(.) > 1] %>% 
+      names %>% 
+      sub(" .+", "", .) %>% 
+      { mutate(expression_data_rows, multi_exp = ifelse(biopsy_id %in% ., 1, 0)) }
+  }
+  
+  expression_data_rows = mark_duplicates_biopsy(expression_data_rows)
+  
+  if(verbose){
+    now_row = nrow(expression_data_rows)
+    print(paste(now_row,"remain after collapsing on biopsy_id"))
+  }
+  
+  # collapse duplicates if any
+  if(any(expression_data_rows$multi_exp == 1) ){
+    if(verbose){
+      rows_total = filter(expression_data_rows,multi_exp==1) %>% nrow()
+      unique_redundant = filter(expression_data_rows,multi_exp==1) %>% pull(biopsy_id) %>% unique()
+      nid = length(unique_redundant)
+      print(paste(rows_total, "LINE 74: rows corresponding to ", nid, "unique biopsies are redundant, dropping redundancy"))
+      print(head(unique_redundant))
+    }
+    original = expression_data_rows %>% filter(!is.na(mrna_sample_id))
+    expression_data_rows = group_by(expression_data_rows, patient_id, biopsy_id) %>% 
+      slice_max(str_detect(protocol, "Ribo"), n=1, with_ties = TRUE) %>% 
+      slice_max(ffpe_or_frozen == "frozen", n=1, with_ties = TRUE) %>% 
+      ungroup()
+    
+    
+    expression_data_rows = mark_duplicates_biopsy(expression_data_rows)
+    
+    if(verbose){
+      now_row = nrow(expression_data_rows)
+      print(paste(now_row,"remain after collapsing on biopsy_id"))
+    }
+
+  }else{
```

Comment:
This is the `else` for `if(any(expression_data_rows$multi_exp == 1) )`, right? I'm not following what happens if the condition is false... what happens here?